src/order/order_iso_nat.lean

```diff
@@ -42,4 +42,9 @@ theorem well_founded_iff_no_descending_seq [is_strict_order α r] :
   N ⟨nat_gt (λ n, (f^[n] ⟨a, na⟩).1) $ λ n,
     by { rw [function.iterate_succ'], apply h }⟩⟩⟩
 
+lemma well_founded_of_well_founded {β : Type*} {s : β → β → Prop}
```

Comment:
This is basically `rel_embedding.well_founded` from `order/rel_iso.lean`, isn't it?

Reply:
Oh, thanks, much better; I don't know how I didn't find this. In fact I see now there's even `rel_hom.well_founded` so I there is no need for either the `is_strict_order` or injectivity assumptions.

Incoming commit.